exchanges/orderbook/orderbook_types.go

```diff
@@ -38,23 +38,23 @@ var (
 	errChecksumStringNotSet = errors.New("checksum string not set")
 )
 
-var service = Service{
-	books: make(map[string]Exchange),
-	Mux:   dispatch.GetNewMux(nil),
+var s = store{
+	orderbooks:  make(map[key.ExchangePairAsset]bookWithExchangeID),
+	exchangeIDs: make(map[string]uuid.UUID),
+	signalMux:   dispatch.GetNewMux(nil),
 }
 
-// Service provides a store for difference exchange orderbooks
-type Service struct {
-	books map[string]Exchange
-	*dispatch.Mux
-	mu sync.Mutex
+type bookWithExchangeID struct {
+	ExchangeID uuid.UUID
+	Depth      *Depth
 }
 
-// Exchange defines a holder for the exchange specific depth items with a
-// specific ID associated with that exchange
-type Exchange struct {
-	m  map[key.PairAsset]*Depth
-	ID uuid.UUID
+// store provides a centralised store for orderbooks
+type store struct {
+	orderbooks  map[key.ExchangePairAsset]bookWithExchangeID
+	exchangeIDs map[string]uuid.UUID
+	signalMux   *dispatch.Mux
+	m           sync.RWMutex
```

Comment:
I think it would be best if you were to write some benchmarks/stats for this. While the RLock will great improve the map check speed. RW mutex is slower for Lock Unlock, which is occurring very very frequently with WS updates.

Since order book is very precious, proving that the RW mutex is the better handler than simply mutex would be good

Reply:
Added bench: [54859a0](https://github.com/thrasher-corp/gocryptotrader/pull/1909/commits/54859a0948d79cdd60619aaa4f464ad88f68e19d)

Just FYI. WS maintains its own lookup system in buffer package and the lock interactions are occurring at the Depth level on both occasions. The reads for this map specifically are >=90% beyond startup and occasional new pair being added. It's just association insertion.

Reply:
From your replies and that I didn't consider everything fully shows me I wasted both our times by reviewing